feat(Tree): create editable Tree node item

leko91 opened the pull request:
This was discussed in this [Slack thread](https://frontify.slack.com/archives/C02PSJTPA3D/p1650997178045199?thread_ts=1650372981.299969&cid=C02PSJTPA3D).

@julianiff
The PR is not complete (test failing, logs, etc.), just want to sync with you first on the implementation, fix any big issues if any, and then complete everything so we skip double work.

julianiff commented:
@leko91 Sorry for the late reply I missed the email. Looks great!

julianiff commented on src/components/Tree/Tree.tsx:
```diff
@@ -16,6 +16,8 @@ export interface TreeFlatListItem {
     value?: string;
     actions?: React.ReactNode[];
     parentId: NullableString;
+    editable?: boolean;
+    onEditableSave?: (value: string) => void;
 }
```
We were discussing about the properties. Do you think it would be enough to just have the Callback property instead of both?

leko91 replied:
It would be, I am also not a fan of having both properties. However, consistency wise, I think editable is a good prop to have.

Guess it's your call in the end 😄

julianiff replied:
Then we leave for now if you see that it helps consistency

julianiff commented on src/components/Tree/components/EditableNodeItem.tsx:
```diff
@@ -0,0 +1,45 @@
+        if (event.key === "Enter") {
+            setShowInput(false);
+
+            onEditableSave && onEditableSave((event.target as HTMLInputElement).value);
```
The interface defined the property not as optional, i think you can remove the check with the &&

julianiff reviewed (approved):
Thx @leko91 for the nice prototype.

leko91 commented:
@julianiff 
Hey Julian, I've updated the code a bit, added a close on blur, focus on opening the input, and written tests. Can you check it out again, please? 😃
